viroconcom/plot.py

```diff
@@ -0,0 +1,520 @@
+    ax : Axes
+        Axes of the figure where the contour should be plotted.
+    contour_label : str, optional (defaults to None)
+        The environmental contour's label that will be used in the legend.
+    x_label : str, optional (defaults to None)
+        Label for the x-axis.
+    y_label : str, optional (defaults to None)
+        Label for the y-axis.
+    line_style : str, optional (defaults to 'b-')
+        Matplotlib line style.
+    alpha : float, optional (default to 1)
+        Alpha value (transparency) for the contour's line.
+    plotted_sample : PlottedSample, optional (defaults to None)
+        The sample that should be plotted and its meta information.
+    x_lim : tuple of floats, optional (defaults to None)
+        x-Axis limit.
+    upper_ylim : float, optional (defaults to None)
+        y-Axis limit.
+    median_x : ndarray of floats, optional (defaults to None)
+        If the median of x2|x1 should be plotted, these are the x-values.
+    median_y : ndarray of floats, optional (defaults to None)
+        If the median of x2|x1 should be plotted, these are the y-values.
+    median_style : str, optional (defaults to 'r-')
+        Matplotlib line style for plotting the median of x2|x1.
+    median_label : str, optional (defaults to 'median of x2|x1')
+        Label for the legend of the plotted median line.
+    """
+
+    # For generating a closed contour: add the first coordinate at the end.
+    xplot = x.tolist()
+    xplot.append(x[0])
+    yplot = y.tolist()
+    yplot.append(y[0])
+
+    # Plot the contour and, if provided, also the sample.
+    if plotted_sample:
+        plot_sample(plotted_sample, ax=ax)
+    if contour_label:
+        ax.plot(xplot, yplot, line_style, alpha=alpha, label=contour_label)
+    else:
+        ax.plot(xplot, yplot, line_style, alpha=alpha)
+    if median_x is not None:
+        ax.plot(median_x, median_y, median_style, label=median_label)
+
+    # Format the figure.
+    if contour_label:
+        plt.legend(loc='upper left', frameon=False)
+    if x_label:
+        plt.xlabel(x_label)
+    if y_label:
+        plt.ylabel(y_label)
+    if x_lim:
+        plt.xlim(x_lim)
+    y_lim_factor = 1.2
+    if plotted_sample and upper_ylim is None:
+        # If there is not enough space for the legend in the upper left corner:
+        # make space for it.
+        max_index = np.where(plotted_sample.y == max(plotted_sample.y))
+        if plotted_sample.x[max_index] < 0.6 * max(max(x), max(plotted_sample.x)):
+            y_lim_factor = 1.35
+
+        upper_ylim = max(max(y), max(plotted_sample.y)) * y_lim_factor
+    elif upper_ylim is None:
+        upper_ylim = max(y) * y_lim_factor
+    plt.ylim((0, upper_ylim))
+
+
+    # Remove axis on the right and on the top (Matlab 'box off').
+    ax.spines['right'].set_visible(False)
+    ax.spines['top'].set_visible(False)
+    ax.yaxis.set_ticks_position('left')
+    ax.xaxis.set_ticks_position('bottom')
+
+
+class PlottedSample():

```

Comment:
Name is confusing to me. This class represents just the data that is necessary to plot one Sample.
An alternative name could be SamplePlotData.
Also consider using a [DataClass](https://docs.python.org/3.7/library/dataclasses.html) (introduced in Python 3.7)